File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/DistrictColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DistrictRow } from "./DistrictRow";

export interface DistrictColumns {
    Id: Column<DistrictRow>;
    Title: Column<DistrictRow>;
    StateTitle: Column<DistrictRow>;
    ShortName: Column<DistrictRow>;
}

export class DistrictColumns extends ColumnsBase<DistrictRow> {
    static readonly columnsKey = 'Masters.District';
    static readonly Fields = fieldsProxy<DistrictColumns>();
}

File: GXpert/GXpert.Web/Modules/Masters/District/DistrictColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Masters.Columns;

[ColumnsScript("Masters.District")]
[BasedOnRow(typeof(DistrictRow), CheckNames = true)]
public class DistrictColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Title { get; set; }
    [QuickFilter]
    public string StateTitle { get; set; }
    public string ShortName { get; set; }
}